ops/cmd/apply_hardforks/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/config"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/fs"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/manage"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/output"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/paths"
+)
+
+func main() {
+	if err := mainErr(); err != nil {
+		output.WriteStderr("%v", err)
+		os.Exit(1)
+	}
+}
+
+func mainErr() error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get working directory: %w", err)
+	}
+
+	output.WriteStderr("working directory: %s", wd)
+
+	superchainDir := path.Join(wd, "superchain")
+	exists, err := fs.DirExists(superchainDir)
+	if err != nil {
+		return fmt.Errorf("error checking for superchain directory: %w", err)
+	}
+	if !exists {
+		return fmt.Errorf("superchain directory does not exist - check working directory")
+	}
+
+	superchains, err := paths.Superchains(wd)
+	if err != nil {
+		return fmt.Errorf("error getting superchains: %w", err)
+	}
+
+	for _, superchain := range superchains {
+		if err := processSuperchainDir(wd, superchain); err != nil {
+			return fmt.Errorf("error processing superchain %s: %w", superchain, err)
+		}
+	}
+
+	return nil
+}
+
+func processSuperchainDir(wd string, superchain string) error {
+	superchainCfgPath := paths.SuperchainConfig(wd, superchain)
+
+	var superchainCfg config.Superchain
+	if err := paths.ReadTOMLFile(superchainCfgPath, &superchainCfg); err != nil {
+		return fmt.Errorf("error reading superchain config: %w", err)
+	}
+
+	cfgs, err := manage.CollectChainConfigs(paths.SuperchainDir(wd, superchain))
+	if err != nil {
+		return fmt.Errorf("error collecting chain configs: %w", err)
+	}
+
+	for _, cfg := range cfgs {
+		chainCfg := cfg.Config
+		if chainCfg.SuperchainTime != nil && int64(*chainCfg.SuperchainTime) < time.Now().Unix() {
```

Comment:
Is it necessary to restrict this to when `SuperchainTime` is in the past? I think chains ought to be able to set it to the the future and things work as expected? I think in general we shouldn't depend on `Now` at all. 

Reference for `superchain_time` semantics:  https://github.com/ethereum-optimism/superchain-registry/blob/main/docs/hardfork-activation-inheritance.md

Reply:
According to the linked doc, this field functions like a hardfork activation timestamp:

> All superchain-wide hardfork activations will be inherited **starting from this timestamp.**

If we don't check `time.Now()` then we will be applying superchain-wide hardfork activations prior to the timestamp in `superchain_time`.